ballerina/listener_endpoint.bal

```diff
@@ -30,8 +30,10 @@ public class Listener {
     # + listenerConfig - Configurations for FTP listener
     # + return - `()` or else an `ftp:Error` upon failure to initialize the listener
     public isolated function init(ListenerConfiguration listenerConfig) returns Error? {
-        self.config = listenerConfig;
-        return initListener(self, self.config);
+        self.config = listenerConfig.clone();
```

Comment:
Just out of curiosity is there any reason to keep this config in a variable ? Is there any other place `config` variable is used other than in the `init`. If we don't have that variable we could remove the `lock` block in line 34 itself.

Reply:
Its used in the start method to get the polling interval